sql/core/src/test/scala/org/apache/spark/sql/SQLQuerySuite.scala

```diff
@@ -808,6 +808,12 @@ class SQLQuerySuite extends QueryTest with SharedSQLContext {
       Row("1"))
   }
 
+  test("SPARK-21588 SQLContext.getConf(key, null) should return null") {
```

Comment:
Maybe directly test against `SQLConf` in `SQLConfSuite`, instead of via `SQLContext`?